src/render/views/CreateEncounterView.tsx

```diff
@@ -30,26 +46,54 @@ export const CreateEncounterView: React.FC<{}> = () => {
     const viewManager = getViewManager();
     const encounterManager = getEncounterManager();
     const [name, setName] = useObservable("");
+    const [actors, setActors] = useObservableArray<Actor>([]);
+    const [speedDialOpen, setSpeedDialOpen] = useObservable(false);
+    const styles = useStyles();
+    const speedDialActions = [
+        { icon: <PersonAddIcon />, name: "Add NPC" },
+        { icon: <PersonAddIcon />, name: "Add NPC From Template" },
+        { icon: <PersonAddIcon />, name: "Add Player Character" },
+        { icon: <PersonAddIcon />, name: "Add All Player Characters" }
+    ];
 
     const create = () => {
         encounterManager.createNewEncounter({
-            actors: [],
+            actors: [...actors.value],
             name: name.value
         });
 
         viewManager.onViewChanged(View.EncounterManagement);
     };
 
+    const addPlayerCharacters = () => {
+        // setActors([
+        //     {
+        //         actorType: ActorType.PC,
+        //         id: "myId",
+        //         initiative: 12,
+        //         name: "plinko",
+        //         template: {}
+        //     }
+        // ]);
+        //actors.concat(db.getActors(playercharacters));
+    };
+
     const cancel = () => {
         viewManager.onViewChanged(View.EncounterManagement);
     };
 
-    const styles = useStyles();
+    const handleSpeedDialClose = () => {
```

Comment:
These are short enough that you can just inline them in the `SpeedDial` itself as `onOpen={() => setSpeedDialOpen(true)}` etc.